executor/substate_provider.go

```diff
@@ -31,13 +33,21 @@ import (
 // ----------------------------------------------------------------------------
 
 // OpenSubstateProvider opens a substate database as configured in the given parameters.
-func OpenSubstateProvider(cfg *utils.Config, ctxt *cli.Context, aidaDb db.BaseDB) Provider[txcontext.TxContext] {
+func OpenSubstateProvider(cfg *utils.Config, ctxt *cli.Context, aidaDb db.BaseDB) (Provider[txcontext.TxContext], error) {
 	substateDb := db.MakeDefaultSubstateDBFromBaseDB(aidaDb)
+
+	if cfg.SubstateEncoding != "" {
```

Comment:
Why do we need to check it? I thought `""` would chose a default option.

Reply:
`cfg.SubstateEncoding` was set to `"default"` by the flags/config system, so manually created config in tests would yield `""`. This was the fastest way to get this version of Substate to work.

Changes where `""` also defensively defaults to `rlp` will be in a separate PR.